### PR TITLE
[FR] Add host family to data path

### PR DESCRIPTION
**Pull request by eric-forte-elastic**
## Summary

Please merge https://github.com/elastic/detection-rules/pull/2810 first! The code for the timestamp fix is included in this PR in order to facilitate testing. 

This PR adds the host family (i.e windows, linux, macos) to the data file path when running `collect-events`. This value is set from the event data for the given host id if it is present, otherwise it prompts the user to supply a host_family. 

## Testing
Run the `collect-events` command with an RTA against an Elastic stack with a host with the Elastic agent running.

Example command: 

```shell
python -m detection_rules es --cloud-id <CLOUD_ID> collect-events --rta-name <RTA_NAME> <HOST_ID>
```

When doing this make sure to execute the RTA on the host as well in order to generate some events to capture. 

<details><summary>Example Output</summary>
<p>
On development machine

```shell

> python -m detection_rules es --cloud-id <CLOUD_ID> collect-events --rta-name adobe_hijack <HOST_ID>

█▀▀▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄   ▄      █▀▀▄ ▄  ▄ ▄   ▄▄▄ ▄▄▄
█  █ █▄▄  █  █▄▄ █    █   █  █ █ █▀▄ █      █▄▄▀ █  █ █   █▄▄ █▄▄
█▄▄▀ █▄▄  █  █▄▄ █▄▄  █  ▄█▄ █▄█ █ ▀▄█      █ ▀▄ █▄▄█ █▄▄ █▄▄ ▄▄█

es_user: <username>
es_password: 
Press any key once detonation is complete ...
3 events saved to: /tmp/detection_rules/detection-rules/unit_tests/data/true_positives/adobe_hijack/windows/endpoint.ndjson
3 events saved to: /tmp/detection_rules/detection-rules/unit_tests/data/true_positives/adobe_hijack/windows/filebeat.ndjson
31 events saved to: /tmp/detection_rules/detection-rules/unit_tests/data/true_positives/adobe_hijack/windows/metricbeat.ndjson
```

On target host machine:

```shell
python -m rta -n adobe_hijack
```

</p>
</details>

### CODE REVIEW

File: detection_rules/utils.py [hunk 1]
```diff
@@ -191,8 +191,16 @@ def unzip_to_dict(zipped: zipfile.ZipFile, load_json=True) -> Dict[str, Union[di
 def event_sort(events, timestamp='@timestamp', date_format='%Y-%m-%dT%H:%M:%S.%f%z', asc=True):
     """Sort events from elasticsearch by timestamp."""
 
+    def _round_microseconds(t: str) -> str:
+        micro_seconds = t.split('.')[-1].split("Z")[0]
+        if t and len(micro_seconds) > 6:
+            micro_seconds = "0." + micro_seconds
+            micro_seconds = str(round(float(micro_seconds), 6)).split(".")[-1]
+            t = t.split('.')[0] + '.' + micro_seconds + "Z"
+        return t
```
Comment (Mikaayenson) [inline detection_rules/utils.py:200]: What do you think about this?

```suggestion
    def round_microseconds(t: str) -> str:
        """Rounds the microseconds part of a timestamp string to 6 decimal places."""

        if not t:
            # Return early if the timestamp string is empty
            return t

        parts = t.split('.')
        if len(parts) == 2:
            # Remove trailing "Z" from microseconds part
            micro_seconds = parts[1].rstrip("Z")

            if len(micro_seconds) > 6:
                # If the microseconds part has more than 6 digits
                # Convert the microseconds part to a float and round to 6 decimal places
                rounded_micro_seconds = round(float(f"0.{micro_seconds}"), 6)
                
                # Format the rounded value to always have 6 decimal places
                # Reconstruct the timestamp string with the rounded microseconds part
                formatted_micro_seconds = f'{rounded_micro_seconds:0.6f}'.split(".")[-1]
                t = f"{parts[0]}.{formatted_micro_seconds}Z"

        return t
```

File: detection_rules/utils.py [hunk 2]
```diff
@@ -191,8 +191,16 @@ def unzip_to_dict(zipped: zipfile.ZipFile, load_json=True) -> Dict[str, Union[di
...
+
     def _event_sort(event):
-        t = event[timestamp]
+        t = _round_microseconds(event[timestamp])
```
Comment (Mikaayenson) [inline detection_rules/utils.py:203]:
```suggestion
       """Calculates the sort key for an event."""
        t = round_microseconds(event[timestamp])
        
        # Return the timestamp in seconds, adjusted for microseconds and then scaled to milliseconds
```

**Review by Mikaayenson — APPROVED** [review]
LGTM: 

We can probably close https://github.com/elastic/detection-rules/pull/2810 if the same typeguard is needed here. 

Nit: added a couple comments to the typeguard.

**Review by terrancedejesus — APPROVED** [review]
Nice, thanks for updating this!

File: detection_rules/utils.py (hunk 1 above)
Comment (eric-forte-elastic) [reply detection_rules/utils.py:200]: I like this, much more readable :+1:

**Pull request merged**